src/lib.rs

```diff
@@ -311,6 +329,64 @@ impl<I: Iterator> Iterator for Pack<I> {
     type Item = Vec<I::Item>;
     fn next(&mut self) -> Option<Self::Item> {
         let item: Vec<_> = self.iter.by_ref().take(self.nb).collect();
-        if item.is_empty() { None } else { Some(item) }
+        if item.is_empty() {
+            None
+        } else {
+            Some(item)
+        }
+    }
+}
+
+/// A builder used to configure the parallele work.
+///
+/// This struct is created by the `with_nb_threads()` method on
+/// `ParIter`.  See its documentation for more.
+#[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
+pub struct ParMapBuilder<I> {
+    iter: I,
+    nb_threads: usize,
+}
+
+impl<I: Iterator> ParMapBuilder<I> {
+    /// # Example
+    ///
+    /// ```
+    /// use par_map::ParMap;
+    /// let a = [1, 2, 3];
+    /// let mut iter = a.iter().cloned().with_nb_threads(2).par_map(|x| 2 * x);
+    /// assert_eq!(iter.next(), Some(2));
+    /// assert_eq!(iter.next(), Some(4));
+    /// assert_eq!(iter.next(), Some(6));
+    /// assert_eq!(iter.next(), None);
+    /// ```
```

Comment:
You can just forward the documentation to the global par_map doc, something as "As par_map, but with a custom number of threads".